content/scripts/build.js

```diff
@@ -1287,7 +1290,7 @@ class Builder {
       tags: metadata.tags || [],
       selective_mode: false,
     };
-    return renderMacros(rawHtml, docEnv);
+    return this.ksRenderer.render(rawHtml, docEnv);
```

Comment:
```suggestion
    return this.ksRenderer.render(rawHtml, docEnv, this.allTitles);
```
???

Reply:
I decided on a different approach for now. If I did it this way, I'd need to either re-generate the `ksRenderer`'s internal data structures each time, which would of course be really inefficient, or develop some way to determine if `this.allTitles` has changed since the last time the the `ksRenderer`'s internal data structures were generated, which although possible (see https://www.npmjs.com/package/object-hash) also seems inefficient since I believe `this.allTitles` never changes after it has been loaded into memory. So I decided on `ksRenderer.use(this.allTitles)` (see the code) within the `ensureAllTitles()` function.